internal/pkg/manifest/validate.go

```diff
@@ -1459,7 +1457,10 @@ func (q SQSQueueOrBool) validate() error {
 	if q.IsEmpty() {
 		return nil
 	}
-	return q.Advanced.validate()
+	if err := q.Advanced.validate(); err != nil {
+		return err
+	}
+	return q.Advanced.FIFO.validate()
```

Comment:
Is there any reason why we prefer this line here over in `func (q SQSQueue) validate`?
```go
func (q SQSQueue) validate() error {
	if q.IsEmpty() {
		return nil
	}
	if err := q.DeadLetter.validate(); err != nil {
		return fmt.Errorf(`validate "dead_letter": %w`, err)
	}
	return q.FIFO.validate()
}
```

Reply:
There is no preference as such, I will move it into `func (q SQSQueue) validate()`